cli/internal/util/env/env_test.go

```diff
@@ -0,0 +1,147 @@
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"REACT_APP_MY_COOL_VAR=cool"}))
+}
+
+func TestGetAllFrameworkEnvPairs_Gatsby(t *testing.T) {
+	t.Setenv("GATSBY_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"GATSBY_MY_COOL_VAR=cool"}))
+}
+
+func TestGetAllFrameworkEnvPairs_Public(t *testing.T) {
+	t.Setenv("PUBLIC_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"PUBLIC_MY_COOL_VAR=cool"}))
+}
+
+func TestGetAllFrameworkEnvPairs_Vue(t *testing.T) {
+	t.Setenv("VUE_APP_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"VUE_APP_MY_COOL_VAR=cool"}))
+}
+
+func TestGetAllFrameworkEnvPairs_Vite(t *testing.T) {
+	t.Setenv("VITE_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"VITE_MY_COOL_VAR=cool"}))
+}
+
+func TestGetAllFrameworkEnvPairs_Redwood(t *testing.T) {
+	t.Setenv("REDWOOD_ENV_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"REDWOOD_ENV_MY_COOL_VAR=cool"}))
+}
+func TestGetAllFrameworkEnvPairs_Sanity(t *testing.T) {
+	t.Setenv("SANITY_STUDIO_MY_COOL_VAR", "cool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+	println("cool")
+	result := GetAllFrameworkEnvPairs()
+	assert.DeepEqual(t, result, []string([]string{"SANITY_STUDIO_MY_COOL_VAR=cool"}))
+}
+
+func TestGetHashableEnvPairs(t *testing.T) {
+	t.Setenv("NEXT_PUBLIC_MY_COOL_VAR", "cool")
+	t.Setenv("NEXT_PUBLIC_MY_COOL_VAR2", "cool1")
+	t.Setenv("NUXT_ENV_MY_COOL_VAR", "cool")
+	t.Setenv("REACT_APP_MY_COOL_VAR", "cool")
+	t.Setenv("GATSBY_MY_COOL_VAR", "cool")
+	t.Setenv("PUBLIC_MY_COOL_VAR", "cool")
+	t.Setenv("PUBLIC_MY_COOL_VAR2", "cool1")
+	t.Setenv("VUE_APP_MY_COOL_VAR", "cool")
+	t.Setenv("VITE_MY_COOL_VAR", "cool")
+	t.Setenv("REDWOOD_ENV_MY_COOL_VAR", "cool")
+	t.Setenv("SANITY_STUDIO_MY_COOL_VAR", "cool")
+	t.Setenv("lowercase", "stillcool")
+	t.Setenv("MY_TEST_VAR", "cool")
+	t.Setenv("12345", "numbers")
+	t.Setenv("one_MORE", "great")
+
+	result := GetHashableEnvPairs([]string{"MY_TEST_VAR", "lowercase", "one_MORE"})
```

Comment:
Some more tests:
- Duplicates
- Empty string
- Mess around with `=` in values.
- *Just* the prefix (e.g. `PUBLIC_`) with no following character.
- Unset keys in the enumerated keys.
- Unicode.

Reply:
👍🏼